### PR TITLE
Add "Obsidian" extension

### PR DESCRIPTION
**Pull request by marcjulianschwarz**
## Description

Add "Obsidian" extension.
This extension lets users interact with notes written in the knowledge management and note taking app Obsidian.
You can search, view, open, edit, copy and paste notes.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/67844154/145678597-cd652295-1f95-4d59-ac0a-80bb561857d8.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)

### CODE REVIEW

File: extensions/obsidian/package.json [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      "type": "checkbox",
+      "title": "Hide Links",
+      "required": false,
+      "description": "Hide links for copying and viewing notes"
+    }
+  ],
+  "commands": [
+    {
+      "name": "index",
+      "title": "Search in Obsidian",
```
Comment (pernielsentikaer) [inline extensions/obsidian/package.json:42]:
```suggestion
      "title": "Search Note",
      "subtitle": "Obsidian",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @marcjulianschwarz 🔥 

Thanks for the contribution, I left some feedback for you to work on 💪 

- Handle errors gracefully when path is wrong

wqELgJk4T8QUVdXL

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by marcjulianschwarz** [comment]
Hi @pernielsentikaer,
thank you for pointing out this issue, I just committed a fix and accepted the change for the commands title and subtitle.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @marcjulianschwarz 🔥 

Thanks for the update, much better! 💪 

A few comments:

- When using the command there is [Empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states), can you look into eliminate this?

- What should "Hide links" do, seems to do nothing here at least, tried markdown link and regular link, it looks the same regardless the setting set.

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by marcjulianschwarz** [comment]
Thank you for the feedback.

I removed the "Empty state flicker" as described in the article.

"Hide links" will remove links in the Wikilink format: `[[Link to another note]]` -> `Link to another note`. This link format is used by nearly everyone in the Obsidian community to connect notes. When copying or viewing notes with the raycast extension these links might be distracting and as they don't include any extra information like a URL they can safely be hidden.

I won't remove normal markdown links as they do contain information like a URL. Additionally these markdown links will display just fine in "Quick Look".

**Review by pernielsentikaer — APPROVED** [review]
Hi @marcjulianschwarz 🔥 

MUCH better, I think we're ready to ship it 🚢 

I'm going to accept it and then will my colleague do a final review before it's finally merged.  😄 

Thank you for your contribution 💯

**Comment by marcjulianschwarz** [comment]
Hi @pernielsentikaer,

thank you for the great review 😊

**Review by mattisssa — APPROVED** [review]

**Comment by marcjulianschwarz** [comment]
Thank you for reviewing and approving the extension.

To update the extension I will have to create new forks with pull requests right?

**Comment by mattisssa** [comment]
Yes, @marcjulianschwarz, you're right 👍

**Pull request merged**